src/components/Tool/FixPhoneNumbers/FixPhoneNumbers.tsx

```diff
@@ -176,51 +176,53 @@ const FixPhoneNumbers: React.FC = () => {
   };
 
   const handleChange = (
-    contactIndex: number,
+    personId: string,
     numberIndex: number,
     event: React.ChangeEvent<HTMLInputElement>,
   ): void => {
-    const temp = [...test];
-    test[contactIndex].numbers[numberIndex].number = event.target.value;
-    setTest(temp);
+    const temp = { ...dataState };
+    dataState[personId].phoneNumbers[numberIndex].number = event.target.value;
+    setDataState(temp);
```

Comment:
Do you like this any better? It's kinda verbose but is immutable...
```suggestion
    setDataState({
      ...dataState,
      [personId]: {
        ...dataState[personId],
        phoneNumbers: [
          ...dataState[personId].phoneNumbers,
          number: event.target.value,
        ],
      },
    });
```

Reply:
It needs an extra layer because I can't set the number like that, I think I would have to split the area where the numberIndex is then do something like

...dataState[personId].phoneNumbers.slice(0, numberIndex),
{...dataState[personId].phoneNumbers[numberIndex], number: event.target,value}
......dataState[personId].phoneNumbers,slice(numberIndex)

if you think the immuatbility is worth it I can add it